compiler/ml/typedtree.mli

```diff
@@ -83,7 +83,8 @@ and pattern_desc =
             See {!Types.row_desc} for an explanation of the last parameter.
          *)
   | Tpat_record of
-      (Longident.t loc * label_description * pattern) list * closed_flag
+      (Longident.t loc * label_description * pattern * bool (* optional *)) list
```

Comment:
optional pattern now explicit

Reply:
For cases like this, as the number of elements in the payload grows, do you think it would make sense to move to inline records at some point?

Reply:
Definitely.
Keeping things self contained for this PR.